### ISSUE TITLE
[BUG] Restarting consensus(internal bug). Report to developers

### ISSUE THREAD

**Issue by matisalimbene** [hyperledger/iroha#5126]
### OS and Environment

Ubuntu

### GIT commit hash

2ffcd00

### Minimum working example / Steps to reproduce

iroha_client_cli -c $IROHA_CLIENT_CONFIG peer register --address (somehost):1347 --key  (somekey)

### Actual result

2024-10-04T15:53:35.782760Z  WARN consensus: iroha_core::sumeragi::main_loop: No block produced in due time, requesting view change... peer_public_key=ed012017D319438B343995848F112BEB38695D3AA696C71257D52B12D630FD0099B792 role=ValidatingPeer
2024-10-04T15:53:36.356154Z ERROR consensus: iroha_core::sumeragi::main_loop: Rotating the entire topology. addr=iroha3:1340
2024-10-04T15:53:36.356201Z ERROR consensus: iroha_core::sumeragi::network_topology: Restarting consensus(internal bug). Report to developers

### Expected result

I expected the transaction to be commited and the new peer registeredcommitted

### Logs

<details>
  <summary>Log contents</summary>
  
  ```json
{
  "peers": 3,
  "blocks": 4,
  "txs_accepted": 5,
  "txs_rejected": 0,
  "uptime": {
    "secs": 609851,
    "nanos": 363000000
  },
  "view_changes": 1,
  "queue_size": 4
}

  ```
</details>

### Notes

I got this error while registering peers using the javascript API, the message mentioning a but and report to developers appears on the Iroha node itself. I get the same error if I execute a peer registration from the command line:

I'm using docker image hyperledger/iroha2:stable-2.0.0-pre-rc.20 , this is the version data extracted from running containers: 
version=2.0.0-pre-rc.20 git_commit_sha="2ffcd00"

```sh
2024-10-04T15:53:35.782760Z  WARN consensus: iroha_core::sumeragi::main_loop: No block produced in due time, requesting view change... peer_public_key=ed012017D319438B343995848F112BEB38695D3AA696C71257D52B12D630FD0099B792 role=ValidatingPeer
2024-10-04T15:53:36.356154Z ERROR consensus: iroha_core::sumeragi::main_loop: Rotating the entire topology. addr=iroha3:1340
2024-10-04T15:53:36.356201Z ERROR consensus: iroha_core::sumeragi::network_topology: Restarting consensus(internal bug). Report to developers
```

Command line sample:
```sh
/config # iroha_client_cli -c $IROHA_CLIENT_CONFIG peer register --address (somehost):1347 --key  (somekey)
Error: 
   0: Failed to register peer
   1: Failed to submit transaction.
   2: deadline has elapsed

Location:
   /rustc/f7ca9df69549470541fbf542f87a03eb9ed024b6/library/core/src/convert/mod.rs:716

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

Instructions get queue, they are not being processed.
```json
{
  "peers": 3,
  "blocks": 4,
  "txs_accepted": 5,
  "txs_rejected": 0,
  "uptime": {
    "secs": 610239,
    "nanos": 525000000
  },
  "view_changes": 1,
  "queue_size": 5
}


```

**Comment by matisalimbene** [comment]
I found that even though I was using the same docker image everywhere `hyperledger/iroha2:stable-2.0.0-pre-rc.20 ` the image pulled from docker compose was different that then one pulled from dockerfile. I was only able to notice because of the different commit_sha displayed at the iroha start up. Also, sometimes I got the "Parity Scale" error that I know it relates to having different Iroha versions running.

I executed the following command:

`docker inspect --format='{{index .RepoDigests 0}}' hyperledger/iroha2:stable-2.0.0-pre-rc.20`

And I was able to confirm that the image where different. I did a docker pull on all the system I had the node runnings and the issue resolved.
